include/soci/once-temp-type.h

```diff
@@ -109,6 +109,48 @@ class prepare_type
 
 } // namespace details
 
+// Note: ddl_type is intended to be used just as once_type_type,
+// but since it can be also used directly (explicitly) by the user code,
+// it is declared outside of the namespace details.
+class SOCI_DECL ddl_type
+{
+public:
+
+    ddl_type(session & s);
+    ddl_type(const ddl_type & d);
+    ddl_type & operator=(const ddl_type & d);
+
+    ~ddl_type() SOCI_NOEXCEPT_FALSE;
+
+    void create_table(const std::string & tableName);
+    void add_column(const std::string & tableName,
+        const std::string & columnName, data_type dt,
+        int precision, int scale);
+    void alter_column(const std::string & tableName,
+        const std::string & columnName, data_type dt,
+        int precision, int scale);
+    void drop_column(const std::string & tableName,
+        const std::string & columnName);
+    ddl_type & column(const std::string & columnName, data_type dt,
+        int precision = 0, int scale = 0);
+    ddl_type & unique(const std::string & name,
+        const std::string & columnNames);
+    ddl_type & primary_key(const std::string & name,
+        const std::string & columnNames);
+    ddl_type & foreign_key(const std::string & name,
+        const std::string & columnNames,
+        const std::string & refTableName,
+        const std::string & refColumnNames);
+
+    ddl_type & operator()(const std::string & arbitrarySql);
+
+    void set_tail(const std::string & tail);
```

Comment:
Would be nice to have a comment explaining what does this do, it's not obvious from just looking at the header.